dask_cuda/device_host_file.py

```diff
@@ -148,6 +148,10 @@ def __init__(
         self.fast = self.host_buffer if memory_limit == 0 else self.host_buffer.fast
 
     def __setitem__(self, key, value):
+        if key in self.device_buffer:
+            # Make sure we register the removal of an existing key
+            del self[key]
```

Comment:
I'm not sure I understand this, what is the case when this is necessary? If the key already exists in `self.device_buffer` but it now refers to a non-CUDA object that's going to be stored in `self.host_buffer`, is that right? If so, does this actually happen in practice?

Reply:
No, in the case of `DeviceHostFile` and in the current Dask/Distributed implementation, I don't think it is likely to happen in practice. 
In `ProxifyHostFile` it is way more likely to happen and I added the check in `DeviceHostFile` for robustness.

Reply:
I see, thanks for confirming.